File: src/main/java/com/linkhogar/application/expense/getExpenseByHome/ExpenseResponse.java
package com.linkhogar.application.expense.getExpenseByHome;

import com.linkhogar.application.expense.GetSplitByExpense.ExpenseSplitResponse;
import com.linkhogar.domain.expense.enums.ExpenseCategory;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.List;
import java.util.UUID;

public record ExpenseResponse(
        UUID id,
        UUID payerId,
        String payerName,
        BigDecimal amount,
        String description,
        ExpenseCategory category,
        LocalDateTime createdAt,
        List<ExpenseSplitResponse> splits
) {}

File: linkhogar-web/src/app/Models/expense/ExpenseResponse.ts
import {ExpenseCategory} from './ExpenseCategory';
import {ExpenseSplitResponse} from './ExpenseSplitResponse';

export interface ExpenseResponse {
  id: string;
  payerId: string;
  amount: number;
  description: string;
  category: ExpenseCategory;
  createdAt: string;
  payerName?: string; // Opcional, por si luego lo traemos del back
  splits: ExpenseSplitResponse[];
}
